compiler/compiler.go

```diff
@@ -1371,7 +1386,21 @@ func (c *Compiler) parseCall(frame *Frame, instr *ssa.CallCommon) (llvm.Value, e
 		default:
 			panic("StaticCallee returned an unexpected value")
 		}
-		return c.parseFunctionCall(frame, instr.Args, targetFunc.LLVMFn, context, targetFunc.IsExported()), nil
+		ret := c.parseFunctionCall(frame, instr.Args, targetFunc.LLVMFn, context, targetFunc.IsExported())
+		retTSSA := instr.Signature().Results()
+		var retT llvm.Type
+		switch retTSSA.Len() {
+		case 0:
+			retT = c.ctx.VoidType()
+		case 1:
+			retT = c.getLLVMType(retTSSA.At(0).Type())
+		default:
+			retT = c.getLLVMType(retTSSA)
+		}
+		if retT != ret.Type() {
+			ret = c.builder.CreateBitCast(ret, retT, "call.cast")
```

Comment:
A bitcast cannot be used for all values, such as structs (used for multiple returns).

Also, please add a test that mismatched return types work with more than one return value.